app/grandchallenge/challenges/models.py

```diff
@@ -676,3 +685,284 @@ def delete_challenge_follows(*_, instance: Challenge, **__):
+    challenge_setup = models.TextField(
+        help_text="Describe the challenge set-up."
+    )
+    data_set = models.TextField(
+        help_text="Describe the training and test datasets you are planning to use."
+    )
+    submission_assessment = models.TextField(
+        help_text="Define the metrics you will use to assess and rank "
+        "participants’ submissions."
+    )
+    challenge_publication = models.TextField(
+        help_text="Please indicate if you plan to coordinate a publication "
+        "of the challenge results."
+    )
+    code_availability = models.TextField(
+        help_text="Will the participants’ code be accessible after the challenge?"
+    )
+    expected_number_of_teams = models.IntegerField(
+        help_text="How many teams do you expect to participate in your challenge?"
+    )
+    average_algorithm_container_size = models.IntegerField(
+        default=10, help_text="Average algorithm container size in GB."
+    )
+    average_number_of_containers_per_team = models.IntegerField(
+        default=10,
+        help_text="Average number of algorithm containers per team.",
+    )
+    inference_time_limit = models.IntegerField(
+        blank=True,
+        null=True,
+        help_text="Time limit per inference job in minutes.",
+    )
+    average_size_of_test_image = models.IntegerField(
+        null=True, blank=True, help_text="Average size of a test image in MB."
+    )
+    phase_1_number_of_submissions_per_team = models.IntegerField(
+        null=True,
+        blank=True,
+        help_text="How many submissions will teams be allowed to make to this phase?",
+    )
+    phase_2_number_of_submissions_per_team = models.IntegerField(
+        null=True,
+        blank=True,
+        help_text="How many submissions will teams be allowed to make to this phase?",
+    )
+    phase_1_number_of_test_images = models.IntegerField(
+        null=True,
+        blank=True,
+        help_text="Number of test images for this phase.",
+    )
+    phase_2_number_of_test_images = models.IntegerField(
+        null=True,
+        blank=True,
+        help_text="Number of test images for this phase.",
+    )
+    number_of_tasks = models.IntegerField(
+        default=1,
+        help_text="If your challenge has multiple tasks, we multiply the "
+        "phase 1 and 2 cost estimates by the number of tasks.",
+    )
+
+    def __str__(self):
+        return self.title
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._orig_status = self.status
+
+    def get_absolute_url(self):
+        return reverse("challenges:request-detail", kwargs={"pk": self.pk})
+
+    def save(self, *args, **kwargs):
+        adding = self._state.adding
+        super().save(*args, **kwargs)
+        if adding:
+            send_challenge_requested_email_to_reviewers(self)
+            send_challenge_requested_email_to_requester(self)
+        if self._orig_status != self.status:
```

Comment:
I would not do this in the same method, maybe move the challenge creation to an admin task?

Reply:
I'm not sure how that would work though. We need the challegne to be created before sending the acceptance email, so separating the challenge creation and email sending doesn't work. I could make the entire status update an admin action, but I think part of the point of this pitch is to avoid that the challenge review is dependent on one of us, right?

Reply:
I see, still I think the best thing to do is make this a separate method on the model. You could then call that automatically from the view when the state changes, or from an admin task if you like. That would still allow changing it from the admin or shell (etc) without automatically having these side effects.